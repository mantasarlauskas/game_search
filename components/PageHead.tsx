import Head from 'next/head';
import { useRouter } from 'next/router';
import { PageTitle } from 'utils/page';
import { Route } from 'utils/routes';

function PageHead({ title }: PageHeadProps) {
    const { route } = useRouter();
    return (
        <Head>
            <title key="title">{title || PageTitle[route as Route]}</title>
            <meta key="viewport" name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        </Head>
    );
}

interface PageHeadProps {
    title?: string | number;
}

export default PageHead;
