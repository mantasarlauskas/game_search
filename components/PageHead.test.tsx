import { render, screen } from '@testing-library/react';
import PageHead from 'components/PageHead';

jest.mock('next/head', () => ({ children }: any) => children);
jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/developers',
        };
    },
}));

describe('<PageHead />', () => {
    it('matches snapshot', () => {
        const { container } = render(<PageHead />);
        expect(container).toMatchSnapshot();
    });

    it('renders page route as title', () => {
        render(<PageHead />);
        expect(screen.getByText('Developers')).toBeInTheDocument();
    });

    it('renders custom title', () => {
        render(<PageHead title="title" />);
        expect(screen.getByText('title')).toBeInTheDocument();
    });
});
