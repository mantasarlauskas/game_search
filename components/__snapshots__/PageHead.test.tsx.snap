// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHead /> matches snapshot 1`] = `
<div>
  <title>
    Developers
  </title>
  <meta
    content="width=device-width, initial-scale=1, maximum-scale=1"
    name="viewport"
  />
</div>
`;
